### DISCUSSION TITLE
Static IP config

### DISCUSSION THREAD

**Discussion by runningman84** [rancher/k3os#796] [General]
It would be great if you would provide some working sample how a static IP could be configured.

Btw. I do not get why you can setup wifi dns and so on quite easily and advice users for a simple static IP to write their own config.

**Comment by axsc** [comment]
I think this is the best solution at the moment.
https://github.com/rancher/k3os/issues/68#issuecomment-492089070

**Comment by f-bn** [comment]
I use this method in the `config.yaml` file, it works well and it's way more readable : 
```
write_files:
- path: /var/lib/connman/default.config
  content: |-
    [service_eth0]
    Type=ethernet
    IPv4=192.168.122.10/255.255.255.0/192.168.122.1
    IPv6=off
    Nameservers=192.168.122.1
```
However, I use this in a VM with only one interface, I didn't test how it goes with multiple interface. I use this documentation to configure it btw : https://manpages.debian.org/stretch/connman/connman-service.config.5.en.html

**Comment by hyacin75** [comment]
> I didn't test how it goes with multiple interface

Works well, just duplicate and give a unique name (service_eth1 for example) but **you have to specific the interface MAC** with MAC=xx:xx:xx:xx:xx:xx or it just applies the last one to all the interfaces.

Thanks!!  Your example got me up and running!

**Comment by JOduMonT** [comment]
what about something like this :

```
sudo connmanctl services
*AO Wired                ethernet_000c29eeaf0e_cable
*AR Wired                ethernet_aae0cd9b094b_cable
*AR Wired                ethernet_da9d68a358a4_cable
*AR Wired                ethernet_d62316791d59_cable

k3os-1185 [~]$ sudo connmanctl services ethernet_000c29eeaf0e_cable
...

k3os-1185 [~]$ sudo connmanctl config ethernet_000c29eeaf0e_cable --ipv4 manual 192.168.116.201 255.255.255.0 192.168.116.2 --nameservers 192.168.116.2

```
credit: https://www.centlinux.com/2019/05/configure-network-on-k3os-machine.html

**Comment by JOduMonT** [comment]
@runningman84 I found this documentation today, which have a part on how to manage the Network
https://github.com/rancher/k3os#networking
